### ISSUE TITLE
TypeError: '<' not supported between instances of 'NoneType' and 'NoneType' in `_sort_summary_list`

### ISSUE THREAD

**Issue by nchaly** [mansenfranzen/autodoc_pydantic#78]
Hey all, I've encountered exception while trying to generate a pydantic models documentation. I believe it is related to a combination of factors (most likely it is related to 'bysource' sorting and usage of pydantic model as a class field):

```
class DataTransformer:
    """Transformer"""

    PARAMS_CLS = DataTransformerConfig  # DataTransformerConfig is a pydantic model
```

conf.py settings:
```
  autodoc_pydantic_model_show_json = True
  autodoc_pydantic_model_show_config = False
  autodoc_pydantic_model_show_config_summary = False
  autodoc_pydantic_model_show_config_member = False
  autodoc_pydantic_model_show_validator_summary = False
  autodoc_pydantic_model_show_validator_members = False
  autodoc_pydantic_model_summary_list_order = 'bysource'  # exception disappears after commenting this line
  autodoc_pydantic_model_member_order = 'bysource'
  autodoc_pydantic_model_signature_prefix = ''
  autodoc_pydantic_validator_list_fields = False
  autodoc_pydantic_settings_show_config_summary = False
  autodoc_pydantic_settings_show_config_member = False
  autodoc_pydantic_field_list_validators = False                               
```

Probably there are other factors, so 've put the complete code base to github: https://github.com/nchaly/autodoc-pydantic-
bug

Versions installed:
- sphinx                    4.3.0
- autodoc-pydantic          1.5.1 

Build command: `sphinx-build -b html docs  docs/_build/html -a -E -T `

Sphinx trace file:
```
    # Sphinx version: 4.3.0
    # Python version: 3.9.7 (CPython)
    # Docutils version: 0.16 release
    # Jinja2 version: 3.0.3
    # Last messages:
    #   updating environment:
    #   [new config]
    #   4 added, 0 changed, 0 removed
    #   reading sources... [ 25%] index
    #   reading sources... [ 50%] pages/generated_refs/app
    #   reading sources... [ 75%] pages/generated_refs/app.data_module
    #   
    #   ====================== slowest reading durations =======================
    #   0.020 index
    #   0.009 pages/generated_refs/app
    # Loaded extensions:
    #   sphinx.ext.mathjax (4.3.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/mathjax.py
    #   sphinxcontrib.applehelp (1.0.2) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/applehelp/__init__.py
    #   sphinxcontrib.devhelp (1.0.2) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/devhelp/__init__.py
    #   sphinxcontrib.htmlhelp (2.0.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/htmlhelp/__init__.py
    #   sphinxcontrib.serializinghtml (1.1.5) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/serializinghtml/__init__.py
    #   sphinxcontrib.qthelp (1.0.3) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/qthelp/__init__.py
    #   alabaster (0.7.12) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/alabaster/__init__.py
    #   sphinx.ext.doctest (4.3.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/doctest.py
    #   sphinx.ext.autodoc.preserve_defaults (1.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/preserve_defaults.py
    #   sphinx.ext.autodoc.type_comment (4.3.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/type_comment.py
    #   sphinx.ext.autodoc (4.3.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py
    #   sphinx.ext.autosummary (4.3.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py
    #   sphinx.ext.napoleon (4.3.0) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/napoleon/__init__.py
    #   sphinxcontrib.autodoc_pydantic (1.5.1) from /usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/__init__.py
    Traceback (most recent call last):
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/cmd/build.py", line 280, in build_main
        app.build(args.force_all, filenames)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/application.py", line 338, in build
        self.builder.build_all()
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 258, in build_all
        self.build(None, summary=__('all source files'), method='all')
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 308, in build
        updated_docnames = set(self.read())
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 415, in read
        self._read_serial(docnames)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 436, in _read_serial
        self.read_doc(docname)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 476, in read_doc
        doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/io.py", line 189, in read_doc
        pub.publish()
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/core.py", line 217, in publish
        self.document = self.reader.read(self.source, self.parser,
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/io.py", line 109, in read
        self.parse()
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/readers/__init__.py", line 77, in parse
        self.parser.parse(self.input, document)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/parsers.py", line 101, in parse
        self.statemachine.run(inputlines, document, inliner=self.inliner)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 170, in run
        results = StateMachineWS.run(self, input_lines, input_offset,
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/statemachine.py", line 241, in run
        context, next_state, result = self.check_line(
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/statemachine.py", line 459, in check_line
        return method(match, context, next_state)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
        self.section(title, source, style, lineno - 1, messages)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 327, in section
        self.new_subsection(title, lineno, messages)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
        newabsoffset = self.nested_parse(
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
        state_machine.run(block, input_offset, memo=self.memo,
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 196, in run
        results = StateMachineWS.run(self, input_lines, input_offset)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/statemachine.py", line 241, in run
        context, next_state, result = self.check_line(
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/statemachine.py", line 459, in check_line
        return method(match, context, next_state)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
        nodelist, blank_finish = self.explicit_construct(match)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
        return method(self, expmatch)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
        return self.run_directive(
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
        result = directive_instance.run()
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/directive.py", line 162, in run
        documenter.generate(more_content=self.content)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 984, in generate
        self.document_members(all_members)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 860, in document_members
        documenter.generate(
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 1774, in generate
        return super().generate(more_content=more_content,
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 984, in generate
        self.document_members(all_members)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 1765, in document_members
        super().document_members(all_members)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 860, in document_members
        documenter.generate(
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 1774, in generate
        return super().generate(more_content=more_content,
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 981, in generate
        self.add_content(more_content)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/directives/autodocumenters.py", line 201, in add_content
        self.add_field_summary()
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/directives/autodocumenters.py", line 300, in add_field_summary
        sorted_fields = self._sort_summary_list(filtered_fields)
      File "/usr/local/Caskroom/miniconda/base/envs/logml/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/directives/autodocumenters.py", line 333, in _sort_summary_list
        return sorted(names, key=sort_func)
    TypeError: '<' not supported between instances of 'NoneType' and 'NoneType'

```

**Comment by mansenfranzen** [comment]
Hi @nchaly,

thanks a lot for reporting this bug at such detailed level. It made it very is to reproduce. 

It is definitely a bug due to an unhandled edge case. I will get back to you soon.

**Comment by mansenfranzen** [comment]
It turns out that the bug is not difficult to fix. However, which is more important, your example reveals a more general question. How should **autodoc_pydantic** document pydantic models/settings which are used as class attributes just like in your example. 

While autodocumenting `DataTransformer` it is not relevant to show every detail of `PARAMS_CLS ` (like json schema, field and validator summary lists etc...). Instead, only a short description is enough. The main focus is on `DataTransformer` in its entirety and not on the most detailed documentation of all its individual potentially complex attributes.

Moreover, when `DataTransformerConfig` is used as an attribute, sphinx automatically adds a link to its main auto-documentation section which allows deep dive inspection. That's yet another reason why a brief description is sufficient.

Hence, I suggest to remove all additional sections such as json schema, field & validator summary lists and config summary from pydantic models/settings if they are documented as an attribute. What do you think? 

This will also automatically resolve the bug you're experiencing.

**Comment by nchaly** [comment]
@mansenfranzen I agree. In fact current behaviour creates a duplicate block of data, which confused me the first time I saw it - it looked at a first glance as if `DataTransformer` itself has now got its own schema.

So yes, a reference to the original pydantic model should be fine.

![image](https://user-images.githubusercontent.com/2665273/147464825-c8a90d11-deff-491a-b32c-690ec0bae3f4.png)

**Comment by mansenfranzen** [comment]
#79 addresses the issue.

Before merging the related PR, it would be great if you could test the bug fix on your site to confirm that my test case did indeed capture the incorrect behavior which you've encountered. To do so, please install the current dev release in your doc-building-environment via `pip install git+git://github.com/mansenfranzen/autodoc_pydantic.git@v1.6.0-a.1` and rebuild your docs.

**Comment by mansenfranzen** [comment]
@all-contributors please add @nchaly for bug and tests

**Comment by mansenfranzen** [comment]
I merged the changes in the v1.6.0 release. There is a [FAQ section](https://autodoc-pydantic.readthedocs.io/en/stable/users/faq.html#document-models-as-an-attribute) in the updated documentation describing your issue. Feel free to reopen or comment. Thanks for reporting this once again.
